source/btle/btle.cpp

```diff
@@ -183,6 +183,7 @@ static void btle_handler(ble_evt_t *p_ble_evt)
                     break;
             }
             nRF5xGap::getInstance().processDisconnectionEvent(handle, reason);
+            // TODO: close pending discoveries
```

Comment:
I believe we should invoke the serviceDiscoveryTermination callback before invoking disconnectionCallback. Not sure yet how to report a failure to service discovery.